### ISSUE TITLE
Gutenboarding: cleanup block UI hiding styles

### ISSUE THREAD

**Issue by simison** [Automattic/wp-calypso#38635]
Once https://github.com/WordPress/gutenberg/pull/18173 is released in NPM together with Gutenberg 7.2 (Wed, Jan 8th), we can remove this bit of CSS and rely on new mechanism instead to hide block UI:

https://github.com/Automattic/wp-calypso/blob/ea170c6cec704da2e86cd9d2129e9b2338ffa886/client/landing/gutenboarding/style.scss#L24-L60

...it was added in https://github.com/Automattic/wp-calypso/pull/37225

**Comment by simison** [comment]
PR to update to latest `@wordpress/*` packages: https://github.com/Automattic/wp-calypso/pull/38554

**Comment by roo2** [comment]
It looks like the changes to gutenberg/block-editor package isn't published to NPM yet as part of 7.2, @ellatrix are you about to update the NPM packages too as part of the 7.2 release? Or are the NPM packages just a separate release task?

https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/package.json (not updated yet)

**Comment by simison** [comment]
I think there was a release 3 days ago.

@roo2 I unassigned you if you don't mind — let's keep this task for a few other new a8c devs.

**Comment by simison** [comment]
Might depend on doing `@wordpress/*` packages update in Calypso first — I haven't confirmed. https://github.com/Automattic/wp-calypso/pull/38880

**Comment by Addison-Stavlo** [comment]
Is the idea here to create our own custom block that utilizes this functionality for its InnerBlocks, and then use that as a wrapper for everything that needs its toolbar hidden?  

I may be be interpreting this wrong, but this doesn't look like an optional prop we can add to blocks to hide UI, but something that needs to be hardcoded as a prop on InnerBlocks inside the blocks edit.js definition (like in the example noted for using it in the Buttons block https://github.com/WordPress/gutenberg/blob/c079120ce39044e53ea41a6a1a246aeec8195433/packages/block-library/src/buttons/edit.js#L20-L25).  If that is the case, having the CSS that is already in place may be the simplest route here?  Or maybe I am missing something?

**Comment by sirreal** [comment]
The idea would be to use the prop added in `@wordpress/block-editor@3.5.0`:

https://github.com/WordPress/gutenberg/pull/18173/files#diff-f3c7dfbd917e42c14ab7c5d2c21b119eR81

 to `BlockList` here:

https://github.com/WordPress/gutenberg/pull/18173/files#diff-f3c7dfbd917e42c14ab7c5d2c21b119eR81

This should let us control the visibility of block UI that we don't require right now rather than hiding these elements via CSS.

We don't have 3.5.0 yet, it's currently blocked. This is the PR: https://github.com/Automattic/wp-calypso/pull/38880

**Comment by yansern** [comment]
The `__experimentalUIParts` attributes doesn't give us the possibility to prevent the block toolbar component from being rendered. We need to add a new attribute called `hasToolbar` to the `__experimentalUIParts` to control whether the block toolbar renders or not render. This will require a PR in Gutenberg.

**Comment by sirreal** [comment]
> We don't have 3.5.0 yet, it's currently blocked. This is the PR: #38880

#38880 has landed.

---

> We need to add a new attribute […] This will require a PR in Gutenberg.

👍 Can you open a PR there? Until we have upstream changes, can we start to use the features of `__experimentalUIParts` in `@wordpress/block-editor@3.5.0` and reduce this CSS hiding right now?

**Comment by yansern** [comment]
> 👍 Can you open a PR there?

Created a PR for `hasToolbar` option which completely prevent the block toolbar from rendering. https://github.com/WordPress/gutenberg/pull/19922

> Until we have upstream changes, can we start to use the features of __experimentalUIParts in @wordpress/block-editor@3.5.0 and reduce this CSS hiding right now?

The 2 options added were `hasSelectedUI` and `hasMovers` as seen here https://github.com/WordPress/gutenberg/pull/18173/files. The `hasSelectedUI` option only prevents the `has-selected-ui` classname from being added to the block. This classname is responsible for hiding/showing the editing border around the block. 

We can't reduce CSS hiding on our part yet because the block toolbar is still being rendered so I hope my PR gets approved soon.

**Comment by yansern** [comment]
I have updated the PR in WordPress/gutenberg#19922 from disabling the BlockToolbar to disabling the entire BlockPopover.

Just to explain the difference between BlockToolbar and BlockPopover component. BlockToolbar is actually a child component of BlockPopover component, the component tree looks like this:

```
BlockPopover
  |_ BlockToolbar
  |_ BlockBreadcrumb
  |_ Inserter (to insert another block)
```

In order to disable the block UI completely, you'll need to use both `hasSelectedUI` and `hasPopover` attributes on the `__experimentalUIParts` property, e.g.

```jsx
<BlockList __experimentalUIParts={ { hasPopover: false, hasSelectedUI: false } } />
```

The `hasPopover` attribute disables the block popover.

The `hasSelectedUI` attribute disables the black border around the block when it is focused.

**Comment by yansern** [comment]
`hasPopover` option is now available in the `3.7.0` of `@wordpress/block-editor. https://github.com/WordPress/gutenberg/commit/b3f900c34d41c282db5a6d5a43f8eaf5f091f3d6

**Comment by yansern** [comment]
This is now fixed in this commit https://github.com/Automattic/wp-calypso/commit/444cc85f7dc0bf7d9d025ce00162c34961476460 in this PR https://github.com/Automattic/wp-calypso/pull/39336.
